src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -167,7 +167,7 @@ public function getForm()
      *
      * @throws BadMethodCallException
      */
-    public function addEventListener($eventName, $listener, $priority = 0)
+    public function addEventListener(string $eventName, $listener, int $priority = 0)
```

Comment:
`callable $listener`?

Reply:
@yceruto , this is to avoid autoloading. See `EventDispatcherInterface`

Reply:
On `EventDispatcherInterface`, this was done because (depending on how you use the dispatcher) you could potentially add 8000 listeners to 5687 different events and only call 50 of them during one request which leaves you with a lot of unnecessary autoload calls.

On forms however, it might be save to add the type-hint.

Reply:
We can add a `callable` typehint if the it is called sooner or later like in `*ChoiceLoader`. Event listeners are often never called.

Reply:
Well, in forms `*_SUBMIT` events are fired less often. So I think it makes sense to do same here.

Reply:
On the other hand, in most cases `*_SUBMIT` listeners are declared in subscribers together with `*_DATA` events which are fired for sure... @xabbuh , could you help us here?

Reply:
I think it should be safe to add the type hint here. The autoloading will only happen for listeners that are registered on forms that are actually used during the current request. And usually you shouldn't have dozens of listeners on a single form.